actions/installer.yml

```diff
@@ -0,0 +1,27 @@
+name: slsa-verifier-installer
+description: 'Installs SLSA verifier and adds it to your PATH'
+branding:
+  icon: 'package'
+  color: 'blue'
+inputs:
+  verifier-release:
```

Comment:
If we follow a model `uses: slsa-framework/slsa-verifier/actions/install@v1.2.3`, then we won't need this option, I think.
I think cosign has this for flexibility and pre-GA / testing. In our case, we don't need to give users this ability. If people really ask, we could add it later. But for now let's keep things simple

Reply:
I think these actions usually download a release rather than build from source at a particular tag but I guess we could make it download v1.2.3 if that's the referenced tag. I'm not sure if it's easy for the action to determine the tag it was referenced by though.

Reply:
Another issue is that we may have to fix bugs in the action after a version is released, but still allow users to download that version of the verifier. i.e. v2.0.0 has a bug in the installer but if we update to v2.0.1 then we have to release an identical version of the verifier as v2.0.1 as well which is a bit awkward.

Reply:
> I think these actions usually download a release rather than build from source at a particular tag but I guess we could make it download v1.2.3 if that's the referenced tag. I'm not sure if it's easy for the action to determine the tag it was referenced by though.

mhh, let's hope we don't need `token-id:write permissions` :/

Reply:
Personally, I think that having the option is ok. Folks can use the latest version of the action and pass what version of the verifier they want to install and limit it to releases (rather than any git ref). The same way that other actions like `setup-go` do.

Reply:
Point taken. Although it may seem awkward to have `@v1.2.3` and an argument saying `version: v1.5.6`. But if we think this is OK, I'm not against it.

BTW, I found a way to get the version without using OIDC: The GitHub context now contains an `action_path` that looks like this:

`"action_path": "/home/runner/work/_actions/laurentsimon/hello-github-actions/v1.0.0/action-a"`, corresponding to the code below in the calling workflow. 
```
uses: laurentsimon/hello-github-actions/action-a@v1.0.0
```

Reply:
Thinking more about this, I think @ianlewis is right about allowing such an option and decoupling the action ref from the version to install. If @asraa is fine with this, let's do that.

Reply:
I kept the ```verifier-release``` parameter and agree we should version the action separately to verifier releases. FWIW, GitHub recommends versioning actions and application code separately: https://docs.github.com/en/actions/creating-actions/about-custom-actions#choosing-a-location-for-your-action.

Reply:
> Another issue is that we may have to fix bugs in the action after a version is released, but still allow users to download that version of the verifier. i.e. v2.0.0 has a bug in the installer but if we update to v2.0.1 then we have to release an identical version of the verifier as v2.0.1 as well which is a bit awkward.

I suppose in this model the verifier and actions would be considered "the same", and a bug in the action is considered a bug in the verifier and users should update. So we would release `v2.0.1` for a fix, and `v2.0.0` could be installed by a user but it would have the bug. Is there a compelling use case for allowing users to keep `v2.0.1`? If they want the v2 release, `v2.0.x` would be latest and always the one they should update to eventually... no? It would not break compatibility either since it's the same MAJOR.

Reply:
Presumably you mean to ask if there's a compelling case for allowing clients to keep ```v2.0.0```, not ```v2.0.1```. I can't think of one. 

I don't have a strong opinion on whether we should version the installer with the binary or separately. I can see pros and cons to either approach. However, if we're going to version them together, do we want the installer to be compatible with older binary versions? If so, do we intend to patch older minor releases to add the installer? That feels like a pretty cumbersome release process.

Reply:
> If so, do we intend to patch older minor releases to add the installer? That feels like a pretty cumbersome release process.

Wdum? If we version them together, each GHA version would only support the same binary version. We would only support versions >= first release of the GHA.

Let me know if I missed something.

Reply:
Hmmm I went back and forth on my opinions on this:

On coupling:
* Good for end user update story -- dependabot/etc will be able to update verifier versions
* I don't think we would support old verifiers if we went this route. 

On decoupling:
* I see many examples of actions that decouple: `golangci-lint-action`, `cosign-installer`, `setup-go`. So there's already a precendent.
* Clients miss out on updates because dependabot won't update args. If there's a bug fix/patch release in the `verifier-version` then how will clients receive the update? Right now the code isn't so stable and I expect this to be moving and important.
* Maybe clients are encouraged to use `latest` release unless they have a good reason not to, anyway.
* This would require a separate repo no? To manage separate versioning. Is there any blockers to doing so?